samples/original-manual-adjusted.rb

```diff
@@ -30,7 +30,7 @@ def dewikify_hi(str, terms, intro = false)
       str = str
         .gsub(CODE_RE) { |x| code << x; "CODE#[#{code.length-1}]" }
         .gsub(/#{Regexp.quote(terms)}/i, '@\0@')
-        .gsub(/CODE#\[(\d+)\]/) { code[$1.to_i] }
+        .gsub(/CODE#\[(\d+)\]/) { code[Regexp.last_match(1).to_i] }
```

Comment:
Hmm, some of these seem ok with `match`, but the if/case statements don't feel good.

e.g.:

```ruby
match_data = ps.match(CODE_RE)
paras.map do |ps|
  if match_data
    yield :code, match_data[1]
  else
    case ps
    when match_data = ps.match(/\A \* (.+)/m)
          yield :list, match_data[1].split(/^ \* /)
```

What do you guys think? Any suggestions on how this could be better written? Or perhaps the cure here is worse than the disease?